src/components/CustomerQuota/NoQuota/NoQuotaCard.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { FunctionComponent, useState } from "react";
+import { differenceInSeconds, format } from "date-fns";
+import { View } from "react-native";
+import { CustomerCard } from "../CustomerCard";
+import { AppText } from "../../Layout/AppText";
+import { color } from "../../../common/styles";
+import { sharedStyles } from "../sharedStyles";
+import { DarkButton } from "../../Layout/Buttons/DarkButton";
+import { Cart } from "../../../hooks/useCart/useCart";
+import { useProductContext } from "../../../context/products";
+import { getAllIdentifierInputDisplay } from "../../../utils/getIdentifierInputDisplay";
+import { usePastTransaction } from "../../../hooks/usePastTransaction/usePastTransaction";
+import { useAuthenticationContext } from "../../../context/auth";
+import { FontAwesome } from "@expo/vector-icons";
+import { formatQuantityText } from "../utils";
+import { styles } from "./styles";
+import { TransactionsByCategory, Transaction } from "./TransactionsByCategory";
+import { ShowFullListToggle } from "./ShowFullListToggle";
+import {
+  DistantTransactionTitle,
+  RecentTransactionTitle,
+  NoPreviousTransactionTitle
+} from "./TransactionTitle";
+import { AppealButton } from "./AppealButton";
+
+const DURATION_THRESHOLD_SECONDS = 60 * 10; // 10 minutes
+const MAX_TRANSACTIONS_TO_DISPLAY = 5;
+
+interface NoQuotaCard {
+  ids: string[];
+  cart: Cart;
+  onCancel: () => void;
+  onAppeal?: () => void;
+}
+/**
+ * Shows when the user cannot purchase anything
+ *
+ * Precondition: Only rendered when all items in cart have max quantity of 0
+ */
+export const NoQuotaCard: FunctionComponent<NoQuotaCard> = ({
+  ids,
+  cart,
+  onCancel,
+  onAppeal
+}) => {
+  const [isShowFullList, setIsShowFullList] = useState<boolean>(false);
+  const { getProduct, allProducts } = useProductContext();
+  const { token, endpoint } = useAuthenticationContext();
+
+  const policyType = cart.length > 0 && getProduct(cart[0].category)?.type;
+
+  const { pastTransactionsResult } = usePastTransaction(ids, token, endpoint);
```

Comment:
Not sure if this was discussed, do you think we should show some spinner while this is loading?

Reply:
Not discussed, will let Eida know about this

Reply:
From discussion with Eida, this will be covered in another story~